README.md

```diff
@@ -2,31 +2,329 @@
+even some quick references of various tokens that can be used with regular
+expressions. This is an extremely helpful tool when you want to write regular
+expressions, and you'd like to test them first before even writing them in your
+code. It is advised you use this tool as you go through these next few lessons
+to help you understand exactly what the regex evaluates to and how to improve
+your own regular expressions when writing them in the future.
 
-A short one or two paragraph summary of the contents of the lessons, recapping
-the learning goals.
+To get started using Regex 101, go ahead and change the "Flavor" to Java 8 since
+we will eventually be using regular expressions with Java. You can also go ahead
+and click on the "gm" in the far right of the "Regular Expression" text box and
+turn off the "multi line" flag for now as well.
+
+Your screen should now look like this:
+
+![Regex-Java-Global](https://curriculum-content.s3.amazonaws.com/java-mod-3/regex/Regex101.png)
+
+Again, try to follow along with the rest of this lesson using Regex 101 to help
+understand the syntax a little bit better.
+
+## Regex Syntax
+
+In its simplest form, a regex can look for an exact literal piece of text. For
+example, if given the sentence:
+
+```text
+Twinkle twinkle little star, how I wonder what you are.
+```
+
+If we were to provide a regular expression of "star", it would find the literal
+word `star` in the above sentence. If we decided to look for the pattern of
+"ar", then it would find the characters "ar" in `star` and `are`.
+
+Pretty simple, huh? Let's look at some metacharacters now to start showing
+just how powerful regular expressions can be.
+
+### Metacharacters
+
+Remember how we learned about escape sequences in the last module in the
+Characters and Strings lesson? Well guess what - our friend, the backslash,
+'\', is coming back to help us with regular expressions! Consider the table
+below of predefined metacharacters:
+
+| Metacharacter | Description                          |
+|---------------|--------------------------------------|
+| `.`           | Matches any character                |
+| `\s`          | Matches any whitespace character     |
+| `\S`          | Matches any non-whitespace character |
+| `\d`          | Matches any digit                    |
+| `\D`          | Matches any non-digit                |
+| `\w`          | Matches any word character           |
+| `\W`          | Matches any non-word character       |
+
+As we can see from the table above, case is crucial - so make sure if you are
+looking for a digit that it is a lowercase 'd' and not an uppercase 'D'!
+
+Let's look at some examples to showcase these metacharacters. In the Regex 101
+tool, provide the following test string:
+
+```text
+September 15th is National Online Learning Day!
+```
+
+#### The dot (.)
+
+The special character `.` will match any single character. Let's enter the
```

Comment:
Not sure if this is worth mentioning but I think it matches every single character _besides line terminators._